examples/tutorials/manage-ntp.markdown

```diff
@@ -0,0 +1,741 @@
+         classes         => results("bundle", "ntp_package");
+
+     debian::
+      "$(ntp_package_name)"   -> { "StandardsDoc 3.2.1" } 
+         policy          => "present",
+         package_module  => apt_get,
+         handle          => "ntp_packages_apt_$(ntp_package_name)",
+         classes         => results("bundle", "ntp_package");
+
+   files:
+    "$(config_file)"
+      create                => "true",
+      handle                => "ntp_files_conf",
+      perms                 => mog( "644", "root", "root" ),
+      template_method       => "inline_mustache",
+      edit_template_string  => "$(config_template_string)",
+      template_data         => mergedata( '{ "driftfile": "$(driftfile)", "servers": servers}' ),
+      classes               => results( "bundle", "ntp_config" );
+
+   services:
+     "$(ntp_service_name)" -> { "StandardsDoc 3.2.2" } 
+       service_policy => "start",
+       classes => results( "bundle", "ntp_service_running" );
+
+    ntp_config_repaired::
+     "$(ntp_service_name)" -> { "StandardsDoc 3.2.2" } 
+       service_policy => "restart",
+       classes => results( "bundle", "ntp_service_config_change" );
+
+
+   reports:
+     ntp_service_running_repaired.inform_mode::
+       "NTP service started";
+
+     ntp_service_config_change_repaired.inform_mode::
+       "NTP service restarted after configuration change";
+
+}
+```
+
+What does this policy do?
+
+Let's review the different sections of the code, starting with the variable declarations which makes use of operating system environment for classification of the time servers.
+
+#### vars
+
+
+```cf3
+   vars:
+     linux::
+       "ntp_package_name" string => "ntp";
+       "config_file" string => "/etc/ntp.conf";
+       "driftfile" string => "/var/lib/ntp/drift";
+       "servers" slist => { "utcnist.colorado.edu", "utcnist2.colorado.edu" };
+
+      # For brevity, and since the template is small, we define it in-line
+       "config_template_string"
+         string => "# NTP Config managed by CFEngine
+driftfile {{{driftfile}}}
+restrict default kod nomodify notrap nopeer noquery
+restrict -6 default kod nomodify notrap nopeer noquery
+restrict 127.0.0.1
+restrict -6 ::1
+{{#servers}} 
+server {{{.}}} iburst
+{{/servers}}
+includefile /etc/ntp/crypto/pw
+keys /etc/ntp/keys
+";
+
+```
+
+A few new variables are defined. The variables `ntp_package_name`, `config_file`, `driftfile`, `servers`, and `config_template_string` are defined under the `linux` context (so only linux hosts will define them). `config_file` is the path to the ntp configuration file, `driftfile` and `servers` are both variables that will be used when rendering the configuration file and `config_template_string` is the template that will be used to render the configuration file. While both `driftfile** and `servers** are set the same for all linux hosts, those variables could easily be set to different values under different contexts.
```

Comment:
`driftfile** and `servers** -- weird markup

Reply:
ugh that happens quite a bit to me in markdown-mode so damned annoying